src/Http/WebUtilities/src/HttpResponsePipeWriter.cs

```diff
@@ -38,122 +39,72 @@ public class HttpResponsePipeWriter : TextWriter
 
         public override void Write(char value)
         {
-            if (_disposed)
-            {
-                throw new ObjectDisposedException(nameof(HttpResponseStreamWriter));
-            }
-
             _singleCharArray[0] = value;
-            var span = new Span<char>(_singleCharArray, 0, 1);
-            Write(span);
+            WriteInternal(_singleCharArray.AsSpan(0, 1));
```

Comment:
Want to see a neat trick? You can get rid of `_singleCharArray` and do this:

```C#
public override void Write(char value)
{
    WriteInternal(MemoryMarshal.CreateSpan(ref value, 1));
}

```

Reply:
🚀neat! thanks